catch/external/Catch2/cmake/Catch2/Catch.cmake

```diff
@@ -184,66 +204,225 @@ endfunction()
 
 ###############################################################################
 
+
+
+
+#------------------------------------------------------------------------------
+# current staging
+function(catch_discover_tests TARGET_LIST TEST_SET)
+  cmake_parse_arguments(
+    ""
+    ""
+    "TEST_PREFIX;TEST_SUFFIX;WORKING_DIRECTORY;TEST_LIST;REPORTER;OUTPUT_DIR;OUTPUT_PREFIX;OUTPUT_SUFFIX"
+    "TEST_SPEC;EXTRA_ARGS;PROPERTIES"
+    ${ARGN}
+  )
+  ## Generate a unique name based on the extra arguments
+  string(SHA1 args_hash "${_TEST_SPEC} ${_EXTRA_ARGS} ${_REPORTER} ${_OUTPUT_DIR} ${_OUTPUT_PREFIX} ${_OUTPUT_SUFFIX}")
+  string(SUBSTRING ${args_hash} 0 7 args_hash)
+  # Define rule to generate test list for aforementioned test executable
+  set(ctest_include_file "${CMAKE_CURRENT_BINARY_DIR}/${TEST_SET}_include-${args_hash}.cmake")
+  set(ctest_tests_file "${CMAKE_CURRENT_BINARY_DIR}/${TEST_SET}_tests-${args_hash}.cmake")
+  file(RELATIVE_PATH ctestincludepath ${CMAKE_CURRENT_BINARY_DIR} ${ctest_include_file})
+  file(RELATIVE_PATH ctestfilepath ${CMAKE_CURRENT_BINARY_DIR} ${ctest_tests_file})
+  file(RELATIVE_PATH _CATCH_ADD_TEST_SCRIPT ${CMAKE_CURRENT_BINARY_DIR} ${ADD_SCRIPT_PATH})
+  file(RELATIVE_PATH CATCH_INCLUDE_PATH ${CMAKE_CURRENT_BINARY_DIR} ${CATCH_INCLUDE_PATH})
+  if(NOT ${CMAKE_VERSION} VERSION_LESS "3.10.0")
+      file(WRITE ${ctest_include_file} "set(exc_names ${TARGET_LIST})\n")
+      file(APPEND ${ctest_include_file} "set(TARGET ${TEST_SET})\n")
+      file(APPEND ${ctest_include_file} "set(_TEST_LIST ${TEST_SET}_TESTS)\n")
+      file(APPEND ${ctest_include_file} "set(ctestfilepath ${ctestfilepath})\n")
+      file(APPEND ${ctest_include_file} "set(_CATCH_ADD_TEST_SCRIPT ${_CATCH_ADD_TEST_SCRIPT})\n")
+      file(APPEND ${ctest_include_file} "set(crosscompiling_emulator ${crosscompiling_emulator})\n")
+      file(APPEND ${ctest_include_file} "set(_PROPERTIES ${_PROPERTIES})\n")
+      file(APPEND ${ctest_include_file} "include(${CATCH_INCLUDE_PATH})\n")
+      # Add discovered tests to directory TEST_INCLUDE_FILES      
+      set_property(DIRECTORY
+        APPEND PROPERTY TEST_INCLUDE_FILES "${ctestincludepath}"
+      )
+  endif()
+
+endfunction()
+
+###############################################################################
+
 set(_CATCH_DISCOVER_TESTS_SCRIPT
   ${CMAKE_CURRENT_LIST_DIR}/CatchAddTests.cmake
   CACHE INTERNAL "Catch2 full path to CatchAddTests.cmake helper file"
 )
 
+
 ###############################################################################
 # function to be called by all tests
-function(hip_add_exe_to_target)
+function(hip_add_exe_to_target_compile_time_detection)
```

Comment:
I dont see this function being used anywhere? am I missing something here?

Reply:
Let us discuss more on this  on the internal review as there are some additional changes planned on top of this patch. To unblock the CI, I am approving and merging this PR.